### ISSUE TITLE
[PostgreSQL] Parameters are swapped

### ISSUE THREAD

**Issue by joernroeder** [Zewo/Zewo#204]
@Danappelxx your quick reminder :)

consider the following code to update a specified user (in my case `foo@foo.com`), and a table with the `json_data` psql field of type `jsonb`.

``` swift
var update = Update("users")
update.set(QualifiedField("json_data"), "{\"valid\": \"json\"}")
update.filter(QualifiedField("email") == identifier)
```

The `connection.logger` logs `UPDATE users SET json_data = $1 WHERE email = $2` during execution, afterwards the update command threw an `badStatus(PostgreSQL.Result.Status.FatalError, "ERROR:  invalid input syntax for type json\nDETAIL:  Token \"foo\" is invalid.\nCONTEXT:  JSON data, line 1: foo...\n")` Error. I've also added `print(parameter)` to https://github.com/Zewo/PostgreSQL/blob/master/Sources/PostgreSQL/Connection.swift#L187 and got:

``` swift
Optional(SQL.Value.string("foo@foo.com"))
Optional(SQL.Value.string("{\"valid\":\"json\"}"))
```

therefore i think `$1` and `$2` are swapped.
running the following query as raw sql works fine in my case:

``` swift
try self.connection.execute("UPDATE users SET json_data = %@ WHERE email = %@", parameters: [
    "{\"valid\":\"json\"}",
    "foo@foo.com"
])
```

**Comment by Danappelxx** [comment]
Thanks for reporting it! I found the cause of the issue but unfortunately I haven't come up with a clean fix yet. Basically, `sqlParameters` need to be correctly ordered which means that valuesByField and predicate parameters all need to have indices. Currently searching for a better solution, will keep you updated.

@formbound if you have some time, would appreciate some help here :)

**Comment by joernroeder** [comment]
you're welcome. I'm wondering why this bug is coming up now cause i'm using the code since the swift 3.0 release and everything went well back then. I've also scanned the commit history but couldn't find a change that looked related to the problem root of the problem.
Thanks for looking into this 👍

**Comment by paulofaria** [comment]
@Danappelxx any updates?

**Comment by Danappelxx** [comment]
Solution, as usual, turned out to be hard to find but easy to apply. Fix has been pushed and tagged 0.14.3. Hopefully everything works after running `swift build --clean=dist` and rebuilding!

I'll keep this open until you can confirm that its been fixed. I'm a little worried that there may be more to this since it came out of nowhere for you.
